### PR TITLE
New feature:  Add "dimmed", so button toggles through normal/dimmed/hidden

### PR DESCRIPTION
**Pull request by netjeff**
Currently the button toggles through "shown/hidden".   Sometimes I like something in between, where I can still see the watched videos, but they are visually deemphasized through dimming.  With this feature the button now toggles through three states including dimming:

button state

Here is what it looks like in the "dimmed" state, in both the YouTube default (light) theme and dark them:

screenshots

A few other minor changes:

- Tweaked button color for default (light) theme to better match the other buttons.

- Updated comment about `HiddenThresholdPercent` as discussed in [#23](https://github.com/EvHaus/youtube-hide-watched/pull/23#discussion_r324506832), that YouTube doesn't seem to track exactly.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -51,16 +54,41 @@
...
+.YT-HWV-WATCHED-DIMMED:after {
+    content: " ";
+    position: absolute;
+    height: 100%;
+    top: 0;
+    left: 0;
+    right: 0;
+    pointer-events: none; /* pointer works on the underlying video */
+    z-index: 9;
+    background: rgba(
```
Comment (EvHaus) [inline main.js]: Why not use `opacity: 0.8` instead?

File: main.js [hunk 2]
```diff
@@ -169,48 +199,55 @@ html[dark] .YT-HWV-BUTTON {
...
-			item.classList.remove('YT-HWV-WATCHED');
+				// Add current class
+				const state = localStorage.YTHWV_STATE;
+				if ('dimmed' === state) {
+					watchedItem.classList.add('YT-HWV-WATCHED-DIMMED');
+				}
+				else if ('hidden' === state) {
+					watchedItem.classList.add('YT-HWV-WATCHED-HIDDEN');
+				}
+				else { // 'normal' or unknown
```
Comment (EvHaus) [inline main.js]: No need for an empty else statement here.

File: main.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // ==UserScript==
 // @name         YouTube: Hide Watched Videos
 // @namespace    https://www.haus.gg/
-// @version      3.0.1
+// @version      4.0
 // @description  Hides watched videos from your YouTube subscriptions page
 // @author       Ev Haus
```
Comment (EvHaus) [inline main.js]: Thanks for your contributions. Feel free to add yourself as a co-author on a new line here!

File: main.js (hunk 2 above)
Comment (netjeff) [reply main.js]: Replaced with just a comment in efd0e372498d1683894b3fabb16fa740d09459ec.  FYI, I hate it when reading someone else's code, or even my own old code, wondering if the author forgot to include a final `else`.  So my habit is to always start with an empty else.

File: main.js (hunk 3 above)
Comment (netjeff) [reply main.js]: Thanks!  Looking at the [Tampermonkey docs](https://www.tampermonkey.net/documentation.php#_author) it wasn't clear if a second `@author` was allowed, so added to the existing tag in efd0e372498d1683894b3fabb16fa740d09459ec.  I also updated the LICENSE.

File: main.js (hunk 1 above)
Comment (netjeff) [reply main.js]: Yeah, using the `opacity`  property of CSS is much simpler.  Great suggestion. 

You may be able to tell that I'm not very experienced with CSS.  :-) FYI, I wasn't sure how to achieve the effect, so googled "css semi transparent overlay" which led me to this [StackOverflow discussion](https://stackoverflow.com/questions/17859993/basic-css-how-to-overlay-a-div-with-semi-transparent-div-on-top).  Knowing now about `opacity` I can see that discussion was about a much more complicated effect.

**Pull request merged**